### PR TITLE
Refactor maps

### PR DESCRIPTION
**Pull request by xi**
This fixes some of the points from #64.

See the individual commit messages for breaking changes.

### CODE REVIEW

File: adhocracy4/maps/mixins.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def dump_geojson(self):
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context['mapitems_json'] = self.dump_geojson()
-        context['map_url'] = settings.BASE_MAP
+        context['baseurl'] = settings.MAP_BASEURL
         context['polygon'] = json.dumps(self.module.settings_instance.polygon)
```
Comment (MagdaN) [inline adhocracy4/maps/mixins.py]: Is this the URL to the tile server? If yes, I find it confusing to call this base_url.

Comment (xi) [inline adhocracy4/maps/mixins.py]: It is the name that was used before in most places. I could not think of a better alternative yet.

**Comment by xi** [comment]
I will also implement the "replace view mixins for list and detail with template tag to use in detail and list view" refactoring. This mitigates the risk of naming collisions in the context.

File: adhocracy4/maps/mixins.py (hunk 1 above)
Comment (MagdaN) [inline adhocracy4/maps/mixins.py]: I think the name should somehow describe, that it is about maps. base_url sounds like host name.

Comment (xi) [inline adhocracy4/maps/mixins.py]: it is always used in the context of maps, so repeating that it is about maps should not be required. That is of course not true for the view mixins. But I am still working on converting them to template tags.

Comment (MagdaN) [inline adhocracy4/maps/mixins.py]: You do what you want, I find it missleading, but you get your way.

File: tests/maps/test_maps_templatetags.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+    match = re.match(expected, helpers.render_template(template, context))
+    assert match
+    _point = match.group('point')
+    assert json.loads(unescape(_point)) == point
+    _polygon = match.group('polygon')
+    assert json.loads(unescape(_polygon)) == area_settings.polygon
+
+
+# @pytest.fixture
```
Comment (vellip) [inline tests/maps/test_maps_templatetags.py]: Shall these stay commented out?

**Comment by MagdaN** [comment]
I am not sure if I like, that the map height is set in CSS, as that way the map does not work out of the box. I would prefer to pass the height to the widget or set a min-height with inline styles.

**Comment by xi** [comment]
> I am not sure if I like, that the map height is set in CSS, as that way the map does not work out of the box.

I reverted my changes for now as I did not find a good way to make it work out of the box and allow simple styling with CSS at the same time. I would prefer merging this PR soon and work on the height separately.

**Pull request merged**